pkg/kubectl/cmd/apply/apply.go

```diff
@@ -350,6 +364,48 @@ func (o *ApplyOptions) Run() error {
 			glog.V(4).Infof("error recording current command: %v", err)
 		}
 
+		if o.ServerSideApply {
+			// Send the full object to be applied on the server side.
+			data, err := runtime.Encode(unstructured.UnstructuredJSONScheme, info.Object)
+			if err != nil {
+				return cmdutil.AddSourceToErr("serverside-apply", info.Source, err)
+			}
+			options := metav1.UpdateOptions{}
+			if o.ServerDryRun {
+				options.DryRun = []string{metav1.DryRunAll}
+			}
+			obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(
+				info.Namespace,
+				info.Name,
+				types.ApplyPatchType,
+				data,
+				&options,
+			)
+			if err == nil {
+				info.Refresh(obj, true)
+				metadata, err := meta.Accessor(info.Object)
+				if err != nil {
+					return err
+				}
+				visitedUids.Insert(string(metadata.GetUID()))
+				count++
+				if len(output) > 0 && !shortOutput {
```

Comment:
Nit: `len(output) > 0` is much harder to read than `output != ""` - because `output != ""` lets me infer the type of `output`.  But I know some people think the opposite - they probably eat their eggs from the wrong end as well.